docs/proposals/api-grouping.md

```diff
@@ -0,0 +1,375 @@
+API Groups Design Proposal
+--------------------------
+
+## Purpose
+
+Goals:
+
+- document current API groups
+- firmly decide what group Job goes into when it goes to stable.
+- suggest (not binding) how other existing and future types might be (re)grouped
+- get feedback on those suggestions
+
+Non-goal:
+
+- Figure out a plan for deprecating or compatibly moving v1 apis to new groups
+- Figure out a plan for moving  `/api/v1` to to under `/apis`.
+
+## Current State
+
+At HEAD as of 25 Jan 2016, we have 6 API groups:
+
+```
+/api/v1
+/apis/extensions/v1beta1
+/apis/extensions/v1beta1
+/apis/abac/v0
+/apis/abac/v1beta1
+/apis/authorization/v1beta1
+/apis/componentconfig/v1alpha1/
+```
+
+Those api groups have nearly 50 "top-level" types: types that are
+visible to users and which you can GET and/or POST by themselves
+(not nested in some other type).
+
+```
+/api/v1
+	persistentVolume
+	persistentVolumeClaim
+	pod
+	podTemplate
```

Comment:
That is, as a top-level resource.

We share the type, embedded in all the controller types.